backend/app/controllers/books_controller.rb

```diff
@@ -2,13 +2,24 @@
 
 # Books Controller
 class BooksController < ApplicationController
+  def index
+    render(json: { data: {
```

Comment:
Last suggestion from me is to use a presenter here. An example of that could be:

render json: BookPresenter.new(books)

The idea here is the concept of using PORO's (Plain Old Ruby Objects) https://sulmanweb.com/plain-old-ruby-objects-poros-in-rails-fat-models/ this is Ruby on Rails' solution to the thin-fat problem of MVC. This allows heavy logic to be in services, presenters, and helpers instead of the "surface" of your code.

Reply:
This is what I'm looking for, separation of business logic. Thanks!

Reply:
No sure if I did [this](https://github.com/ronsigter/bookshelf/pull/9/files#diff-db6d690fc836206347304804f153dd39ba30febb6dea74958ed07ef3b149dcd8R1) right.

Reply:
This is good enough. In practice though you'd want to be more specific to what you're presenting like BookPresenter and cite the columns you want to show. This is because some columns in some collections are private or you don't need to show them at all. You can also add the columns as a parameter to make the CollectionPresenter generic, but this is good for now as I just wanted you to do the presenter :)